### ISSUE TITLE
doesn't detect imports from implicit namespace packages (PEP 420)

### ISSUE THREAD

**Issue by npip99** [DetachHead/basedpyright#1250]
### Description

Currently, I have these files:

```
your-repo/
├── ml/
│   └── ... ml code
├── scripts/
│   └── ... scripts code
├── pyproject.toml
```

With `pyproject.toml` containing
```
[project]
name = "ml"
...
[tool.setuptools.packages.find]
where = ["."]
include = ["ml", "scripts"]
```

However, imports from the scripts doesn't work correctly,

![Image](https://github.com/user-attachments/assets/bd5d5167-9c31-4469-90e4-e95c21b01fbb)

On the other hand, imports from ml directory work fine. Also, executing `touch ./scripts/__init__.py` fixes it. But, `__init__.py` shouldn't be necessary based on [PEP 420](https://peps.python.org/pep-0420/#specification). This can also be observed by actually running the script; at runtime, it imports fine, so it's only an issue with the typechecker.

## Workaround

Do `touch ./other_directory/__init__.py`

## Ideal

It works without `__init__.py`

**Comment by DetachHead** [comment]
i can't reproduce this. could you provide some more info please:

- what version of basedpyright you're using
- is basedpyright running from the pypi package in your environment, or the version that comes bundled with the vscode extension?
- does the same thing happen in pyright or pylance?
- the contents of the `scripts` and `ml` directories (the `ml` directory doesn't seem to be used in your example at all, is it actuaally needed as part of this minimal repro?)
- what tool(s) you're using to manage your python environment (eg. uv, pip/virtualenv, etc)

**Comment by DetachHead** [comment]
closing this. if you are still experiencing this issue please provide the additional info mentioned above and we can re-open
